### ISSUE TITLE
lines starting with curly braces (destructuring) - let's improve the error message

### ISSUE THREAD

**Issue by revelt** [eslint/eslint#9291]
**Please describe what the rule should do:**

> "The parentheses around the [destructuring] expressions are necessary because statements must not begin with curly braces in JavaScript" – http://exploringjs.com/es6/ch_destructuring.html

Currently, the lines that _do start_ with a curly brace just cause a parsing error with description: 
`Parsing error: Unexpected token =`

Consider this code (excuse my contrived example's logic):

```js
var testObj = { someSetting: 1 };
var someSetting;
{ someSetting } = testObj; // <---- this line would cause this fatal parsing error
```
<img width="430" alt="screen shot 2017-09-12 at 12 32 54" src="https://user-images.githubusercontent.com/8344688/30323885-dc5b8768-97b6-11e7-82ff-601e2b8f10a2.png">

What if we improved ESLint to actually say that lines can't start with a curly brace and suggest wrapping it with parentheses?

The current fatal error message is not very user-friendly and does not give a clue how to solve it.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Why should this rule be included in ESLint (instead of a plugin)?**

Because it is a parser error and it doesn't suggest any solution.

**Comment by not-an-aardvark** [comment]
Thanks for the report. This error message comes from [acorn](https://github.com/ternjs/acorn), so I would recommend opening an issue there if you'd like to see a change.

Keep in mind that in general, it's difficult for parsers to determine what the user was trying to do when it encounters invalid syntax. For example, changing the `=` to a `;` (or simply removing the `=`) would also be a valid way to fix the syntax error, and the parser doesn't know which version you meant. So it's not clear that the updated error message you're proposing would actually be better in all cases.

**Comment by not-an-aardvark** [comment]
Closing because there isn't really anything we can do within ESLint to improve this. Feel free to create an issue on [acorn](https://github.com/ternjs/acorn) if you'd like to see this changed.

**Comment by revelt** [comment]
That's cool. Thanks!
